drivers/spi/spi_nrfx_spis.c

```diff
@@ -127,14 +137,34 @@ static int prepare_for_transfer(const struct device *dev,
 		return -EINVAL;
 	}
 
+	err = dmm_buffer_out_prepare(dev_config->mem_reg, tx_buf, tx_buf_len, (void **)&tx_buf);
+	if (err != 0) {
+		LOG_ERR("DMM TX allocation failed err=%d", err);
+		goto out_alloc_failed;
+	}
+
+	err = dmm_buffer_in_prepare(dev_config->mem_reg, rx_buf, rx_buf_len, (void **)&rx_buf);
```

Comment:
Are you sure it's good to potentially change the `tx_buf` and `rx_buf` pointers here in such a convoluted manner?

Reply:
Used local variables. It should increase readability.